### PR TITLE
AI error messages

### PR DESCRIPTION
**Pull request by LunarX**
Depends on #1392

### CODE REVIEW

File: app/src/main/res/values-fr/strings.xml [hunk 1]
```diff
@@ -51,6 +51,10 @@
     <string name="addUnknownRecipientTitle">Ajouter un destinataire</string>
     <string name="aiButtonInsert">Insérer</string>
     <string name="aiButtonRefine">Affiner</string>
+    <string name="aiButtonRetry">Réessayer</string>
+    <string name="aiErrorMaxTokenReached">Votre demande est trop longue, veuillez réessayer sous un format plus court.</string>
+    <string name="aiErrorTooManyRequests">Il semblerait que vous avez effectué trop de requêtes en un court laps de temps. \n\nVeuillez patienter quelques instants.</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-fr/strings.xml:56]:
```suggestion
    <string name="aiErrorTooManyRequests">Il semblerait que vous avez effectué trop de requêtes en un court laps de temps.\n\nVeuillez patienter quelques instants.</string>
```
Change it on Loco too.

File: app/src/main/res/values-de/strings.xml [hunk 2]
```diff
@@ -51,6 +51,10 @@
     <string name="addUnknownRecipientTitle">Einen Empfänger hinzufügen</string>
     <string name="aiButtonInsert">Einfügen</string>
     <string name="aiButtonRefine">Verfeinern</string>
+    <string name="aiButtonRetry">Wiederholen</string>
+    <string name="aiErrorMaxTokenReached">Ihre Anfrage ist zu lang, bitte versuchen Sie es noch einmal in einem kürzeren Format.</string>
+    <string name="aiErrorTooManyRequests">Es scheint, dass Sie in kurzer Zeit zu viele Suchanfragen gestellt haben. \n\nBitte haben Sie einen Moment Geduld.</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-de/strings.xml:56]:
```suggestion
    <string name="aiErrorTooManyRequests">Es scheint, dass Sie in kurzer Zeit zu viele Suchanfragen gestellt haben.\n\nBitte haben Sie einen Moment Geduld.</string>
```
Change it on Loco too.

File: app/src/main/res/values-es/strings.xml [hunk 3]
```diff
@@ -51,6 +51,10 @@
     <string name="addUnknownRecipientTitle">Añadir un destinatario</string>
     <string name="aiButtonInsert">Inserte</string>
     <string name="aiButtonRefine">Afinar</string>
+    <string name="aiButtonRetry">Reintentar</string>
+    <string name="aiErrorMaxTokenReached">Su solicitud es demasiado larga. Por favor, inténtelo de nuevo en un formato más corto.</string>
+    <string name="aiErrorTooManyRequests">Parece que ha realizado demasiadas solicitudes en un corto espacio de tiempo. \n\nPor favor, espere unos momentos.</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-es/strings.xml:56]:
```suggestion
    <string name="aiErrorTooManyRequests">Parece que ha realizado demasiadas solicitudes en un corto espacio de tiempo.\n\nPor favor, espere unos momentos.</string>
```
Change it on Loco too.

File: app/src/main/res/values-it/strings.xml [hunk 4]
```diff
@@ -51,6 +51,10 @@
     <string name="addUnknownRecipientTitle">Aggiungi un destinatario</string>
     <string name="aiButtonInsert">Inserisci</string>
     <string name="aiButtonRefine">Affinare</string>
+    <string name="aiButtonRetry">Riprova</string>
+    <string name="aiErrorMaxTokenReached">La richiesta è troppo lunga. Riprovare con un formato più breve.</string>
+    <string name="aiErrorTooManyRequests">Sembra che abbiate fatto troppe richieste in un breve lasso di tempo. \n\nLa preghiamo di attendere qualche istante.</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-it/strings.xml:56]:
```suggestion
    <string name="aiErrorTooManyRequests">Sembra che abbiate fatto troppe richieste in un breve lasso di tempo.\n\nLa preghiamo di attendere qualche istante.</string>
```
Change it on Loco too.

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -57,6 +57,10 @@
     <string name="addUnknownRecipientTitle">Add a recipient</string>
     <string name="aiButtonInsert">Insert</string>
     <string name="aiButtonRefine">Refine</string>
+    <string name="aiButtonRetry">Retry</string>
+    <string name="aiErrorMaxTokenReached">Your request is too long, please try again in a shorter format.</string>
+    <string name="aiErrorTooManyRequests">It seems you’ve made too many requests in a short amount of time. \n\nPlease wait a few moments.</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values/strings.xml:62]:
```suggestion
    <string name="aiErrorTooManyRequests">It seems you’ve made too many requests in a short amount of time.\n\nPlease wait a few moments.</string>
```
Change it on Loco too.

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -57,6 +57,10 @@
...
+    <string name="aiErrorUnknown">An error has occurred. If this error persists, please contact our helpdesk. \n\nThank you for your patience.</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values/strings.xml:63]:
```suggestion
    <string name="aiErrorUnknown">An error has occurred. If this error persists, please contact our helpdesk.\n\nThank you for your patience.</string>
```
Change it on Loco too.

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/AiPropositionFragment.kt [hunk 7]
```diff
@@ -146,29 +152,79 @@ class AiPropositionFragment : Fragment() {
...
-                    buttonLayout.isVisible = true
-                    generationLoader.isGone = true
+                    propositionTextView.text = body
+                    setUiVisibilityState(UiState.PROPOSITION)
                 }
-                PropositionStatus.ERROR -> {
-                    // TODO
-                    newMessageViewModel.snackBarManager.setValue(getString(RCore.string.anErrorHasOccurred))
+                PropositionStatus.ERROR,
+                PropositionStatus.MAX_TOKEN_EXCEEDED,
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/AiPropositionFragment.kt:159]:
```suggestion
                PropositionStatus.MAX_TOKENS_EXCEEDED,
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/AiPropositionFragment.kt [hunk 8]
```diff
@@ -138,37 +144,106 @@ class AiPropositionFragment : Fragment() {
...
+        setUiVisibilityState(UiState.ERROR, status)
+    }
+
+    private fun setUiVisibilityState(state: UiState, errorType: PropositionStatus? = null) {
+        when (state) {
+            UiState.LOADING -> displayLoadingVisibility()
+            UiState.PROPOSITION -> displayPropositionVisibility()
+            UiState.ERROR -> displayErrorVisibility(errorType)
+        }
+    }
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/AiPropositionFragment.kt:194]:
```suggestion
    private fun setUiVisibilityState(state: UiState, errorType: PropositionStatus? = null) = when (state) {
        UiState.LOADING -> displayLoadingVisibility()
        UiState.PROPOSITION -> displayPropositionVisibility()
        UiState.ERROR -> displayErrorVisibility(errorType)
    }
```

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/AiPropositionFragment.kt [hunk 9]
```diff
@@ -138,37 +144,106 @@ class AiPropositionFragment : Fragment() {
...
+
+        fun setAdditionalErrorStyle() {
+            val displayRetryButton = errorType == PropositionStatus.PROMPT_TOO_LONG
+
+            retryButton.isVisible = displayRetryButton
+            buttonLayout.isGone = displayRetryButton
+            if (!displayRetryButton) {
+                insertPropositionButton.isEnabled = false
+                refineButton.isGone = true
+            }
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/AiPropositionFragment.kt:231]: Why this if? 

Wouldn't it work like this: 
```suggestion
        insertPropositionButton.isEnabled = displayRetryButton
        refineButton.isVisible = displayRetryButton
```

**Review by FabianDevel — CHANGES_REQUESTED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**